File: 프로그래머스/unrated/155652. 둘만의 암호/둘만의 암호.py
def solution(s, skip, index):
    arr=[]
    skip_len=len(skip)
    s_len=len(s)
    answer = ''
    for i in range(skip_len):
        arr.append(ord(skip[i]))
    for i in range(s_len):
        tmp=ord(s[i])
        idx=0;
        while True:
            if idx==index:
                break
            tmp = tmp + 1
            if tmp in arr:
                continue
            if tmp==123:
                tmp=96
                continue
            idx+=1
        answer = answer+chr(tmp)
    return answer

File: 프로그래머스/unrated/155652. 둘만의 암호/둘만의 암호.java
import java.util.ArrayList;
import java.util.List;
class Solution {
    public String solution(String s, String skip, int index) {
       List<Integer> list = new ArrayList<>();
		String ans = "";
		for(int i=0;i<skip.length();i++) {
			list.add((int)skip.charAt(i));
		}
		int tmp;
		int idx=0;
		for(int i=0;i<s.length();i++) {
			idx=0;
			tmp = (int)s.charAt(i);
			while(true) {
				if(idx==index) {
					break;
				}
				tmp+=1;
				if(list.contains(tmp)) {
					continue;
				}
				if(tmp==123) {
					tmp=96;
					continue;
				}
				idx++;
			}
			ans = ans+String.valueOf((char)tmp);
		}
		return ans;
    }
}